components/nexus-repository/src/main/java/org/sonatype/nexus/repository/search/SearchFacet.java

```diff
@@ -15,17 +15,32 @@
 import org.sonatype.nexus.repository.Facet;
 
 /**
- * ???
+ * Search {@link Facet}, providing means to index/de-index/reindex components.
  *
  * @since 3.0
  */
 @Facet.Exposed
 public interface SearchFacet
-  extends Facet
+    extends Facet
 {
+
+  /**
+   * Index component with specified id.
+   *
+   * @param vertexId id of component to be indexed
+   */
   void index(Object vertexId);
```

Comment:
why doesn't this take an OrientVertex?  I assume that when a component is added, we have this data already, why force this to look it up again?